src/app/pages/receive/components/receive-tokens.tsx

```diff
@@ -1,28 +1,63 @@
+import { useMemo } from 'react';
+
 import { HomePageSelectors } from '@tests/selectors/home.selectors';
 import { css } from 'leather-styles/css';
 import { Stack } from 'leather-styles/jsx';
 
+import { isDefined } from '@shared/utils';
+
 import { copyToClipboard } from '@app/common/utils/copy-to-clipboard';
 import { useToast } from '@app/features/toasts/use-toast';
+import { useAlexSwap } from '@app/pages/swap/hooks/use-alex-swap';
+import { sortSwappableAssetsBySymbol } from '@app/pages/swap/swap.utils';
+import { Avatar, defaultFallbackDelay, getAvatarFallback } from '@app/ui/components/avatar/avatar';
+import { Brc20AvatarIcon } from '@app/ui/components/avatar/brc20-avatar-icon';
 import { BtcAvatarIcon } from '@app/ui/components/avatar/btc-avatar-icon';
+import { RunesAvatarIcon } from '@app/ui/components/avatar/runes-avatar-icon';
+import { Src20AvatarIcon } from '@app/ui/components/avatar/src20-avatar-icon';
 import { StxAvatarIcon } from '@app/ui/components/avatar/stx-avatar-icon';
 
 import { receiveTabStyle } from '../receive-dialog';
 import { ReceiveItem } from './receive-item';
 
+interface ReceivableAsset {
+  address: string;
+  fallback: string;
+  icon: string;
+  name: string;
+}
+
 interface ReceiveTokensProps {
   btcAddressNativeSegwit: string;
+  btcAddressTaproot: string;
   stxAddress: string;
   onClickQrBtc(): void;
   onClickQrStx(): void;
 }
 export function ReceiveTokens({
   btcAddressNativeSegwit,
+  btcAddressTaproot,
   stxAddress,
   onClickQrBtc,
   onClickQrStx,
 }: ReceiveTokensProps) {
   const toast = useToast();
+  const { createSwapAssetFromAlexCurrency, supportedCurrencies } = useAlexSwap();
+
+  const receivableAssets: ReceivableAsset[] = useMemo(() => {
+    const swappableAssets = sortSwappableAssetsBySymbol(
+      supportedCurrencies.map(createSwapAssetFromAlexCurrency).filter(isDefined)
+    );
+    return swappableAssets
+      .filter(asset => asset.name !== 'STX')
+      .map(asset => ({
+        address: asset.principal.split('.')[0],
```

Comment:
This is incorrect, unless I am missing something about what we are doing here?

Reply:
No, you are right, I just didn't understand how it worked and that looked like an address to me. I'll change it so all of these assets use the stacks fungible address here. 

Just to confirm, is that just the regular stacks address from:
```
 const stxAddress = useCurrentAccountStxAddressState();
 ```

So essentially all of these tokens will be received by sending them to the users standard current stacks address

Reply:
Yep, that is correct.

Reply:
For reference, you can see here when the contract is deployed for a new token on Stacks, the `Contract` listed here before the `.` is the contract `principal` (address).

https://explorer.hiro.so/txid/SP1AY6K3PQV5MRT6R4S671NWW2FRVPKM0BR162CT6.leo-token?chain=mainnet

Reply:
Let's make sure to QA this with Ledger. What if the user has a Ledger wallet without Stacks addresses?

I encourage not using `useCurrentAccountStxAddressState` and instead using the StacksAccountLoader

Reply:
Thanks Fara. I just thought they needed to be received to the principal address when trying to figure it out. 

I've updated it now. I can test it with ledger also

Reply:
I tested it on Ledger with only BTC connected and its working OK. It just doesn't show the ALEX tokens
![Uploading Screenshot 2024-04-16 at 16.18.55.png…]()